### PR TITLE
feat(gh-203): Add support for BigQuery.

### PR DESCRIPTION
**Pull request by yingw787**
# Overview

- Mappings between INT64 and INT, BIGINT, SMALLINT, TINYINT with unit tests
- Mappings between FLOAT64 and FLOAT, DOUBLE with unit tests
- Mappings between NUMERIC and DECIMAL with unit tests
- Mappings between STRING and TEXT with unit tests

Haven't added support for other types of casting (e.g. CHAR/VARCHAR/BYTES/TIMESTAMPTZ) due to additional overhead.

Reference: https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types

### CODE REVIEW

File: tests/test_dialects.py [hunk 1]
```diff
@@ -307,6 +307,70 @@ def test_starrocks(self):
             write="starrocks",
         )
 
+    def test_bigquery(self):
+        self.validate(
+            "SELECT CAST(`a`.`b` AS INT) FROM foo",
+            "SELECT CAST(`a`.`b` AS INT64) FROM foo",
+            read="bigquery",
```
Comment (tobymao) [inline tests/test_dialects.py:314]: get rid of all reads

File: tests/test_dialects.py [hunk 2]
```diff
@@ -307,6 +307,70 @@ def test_starrocks(self):
...
+            write="bigquery",
+        )
+
+        self.validate(
```
Comment (tobymao) [inline tests/test_dialects.py:318]: you didn’t test any inverses can you read bigquery int64 and output generic?

File: tests/test_dialects.py (hunk 1 above)
Comment (yingw787) [reply tests/test_dialects.py]: done

File: tests/test_dialects.py (hunk 2 above)
Comment (yingw787) [reply tests/test_dialects.py:317]: That's a good point, I can add that, had some questions:

- What's generic `sqlglot` SQL? Just leave `write=` unspecified?
- When I add `read='bigquery'` using ```"SELECT CAST(`a`.`b` AS INT64) FROM foo",```, I get this error:

```sql
sqlglot.errors.ParseError: Expected TYPE after CAST. Line 1, Col: 24.
  SELECT CAST(`a`.`b` AS INT64) FROM foo
```

I'm not sure how to modify my Dialect class to support the inverse operation, all of the `type_mapping` class attributes have the generic `exp.DataType.Type.*` on the left-hand side and the string on the right-hand side. Do you have any suggestions for me?

Comment (tobymao) [reply tests/test_dialects.py:317]: yes, just leave write unspecified, it should automatically support the inverse

**Pull request merged**